src/SwiftReflector/Demangling/Decomposer.cs

```diff
@@ -0,0 +1,1175 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using SwiftReflector.ExceptionTools;
+using System.Text;
+
+namespace SwiftReflector.Demangling {
+	public class Decomposer {
+		const string kSwiftID = "__T";
+		public const string kSwift4ID = "__T0";
+		public const string kSwift4xID = "_$s";
+		public const string kSwift5ID = "_$S";
```

Comment:
As we discussed in the meeting, I think it's reasonable to shed everything before swift 5 and throw if we get anything else.

Reply:
Thanks, removed.